### ISSUE TITLE
Weird white screen issue

### ISSUE THREAD

**Issue by rainerborene** [zeit/hyper#1042]
- **OS version and name**: elementary OS 0.4 Loki
- **Link of a [Gist](https://gist.github.com/) with the contents of your .hyper.js**: https://github.com/rainerborene/dotfiles/blob/master/.hyper.js

## Issue

Sometimes when I type a command that does not exist it shows that blank page.

![screenshot from 2016-11-25 12 57 21](https://cloud.githubusercontent.com/assets/101501/20628813/5e6b5aa4-b30f-11e6-9c77-0b06588fad47.png)

Is it possible to disable web preview stuff?

**Comment by thejmazz** [comment]
heh, I also have random white screen issues (esp. when using tmux). you can hide it by doing

```javascript
store.dispatch({
  type: 'SESSION_URL_UNSET',
  uid: store.getState().sessions.activeUid
})
```

if you want to disable all web stuff,
just intercept the `SESSION_URL_SET` action in a plugin middleware and don't call `next`:

```javascript
// ~/.hyper_plugins/disableweb/index.js

exports.middleware = (store) => (next) => (action) => {
  if (action.type !== 'SESSION_URL_SET') {
    next(action)
  } else {
    console.log('Intercepted SESSION_URL_SET')
  }
}
```

then add `disableweb` to your local plugins in `~/.hyper.js`. You can use cmd+r to refresh/reload plugins while the terminal is open.

This is the same pattern I used, but the other way around, in my `imgcat` plugin:

https://github.com/thejmazz/hyper-imgcat/blob/6d9410a62f378ec2d62daf55d943968ac3f690fb/index.js#L1-L22

**Comment by ppot** [comment]
@rainerborene Does your command include a URL?

**Comment by rainerborene** [comment]
@ppot not

**Comment by jrop** [comment]
I am running into this when entering a non-existent command within TMUX.  For example, if I enter the command "asdf", then the `isUrlCommand(shell, data) {...}` function erroneously interprets `data` as a URL:

```js
//
// Variable capture from within isUrlCommand(shell, data) {...}
//
shell = "zsh"
data = "\u001b[1;84r\u001b[H\r\nzsh: command not found: asdf\u001b[1;85r\u001b[H\u001b[2B"
matcher = "zsh: (?:(?:command not found)|(?:no such file or directory)): ((?:https?:\/\/)|(?:file:\/\/)|(?:\/\/))?([^\n]+)"
match = ["zsh: command not found: asdf\u001b[1;85r\u001b[H\u001b[2B", undefined, "asdf\u001b[1;85r\u001b[H\u001b[2B"]
protocol = undefined
path = "asdf\u001b[1;85r\u001b[H\u001b[2B"
domain = "asdf\u001b[1;85r\u001b[H\u001b[2B"
result = "1"
return "http://1"
```

As you can see, isUrlCommand is erroneously interpreting the string as being a URL.  Can this issue be opened back up?

**Comment by jrop** [comment]
It seems this was fixed in this commit: b2f67c7

If I clone the repo and run `npm run dev`/`npm run app`, this problem is fixed.  When will this be released as an official build?  This is the only thing keeping me from using Hyper.js right now.

**Comment by ppot** [comment]
Yes @jrop  I fixed it in #1259

**Comment by rauchg** [comment]
URL handling has been vastly [improved on `1.0.1`](https://twitter.com/rauchg/status/815010770618437632)! Thanks for the reports 🙏
